cpp/src/arrow/compute/util.h

```diff
@@ -89,16 +89,23 @@ class ARROW_EXPORT TempVectorStack {
   Status Init(MemoryPool* pool, int64_t size) {
     num_vectors_ = 0;
     top_ = 0;
-    buffer_size_ = PaddedAllocationSize(size) + kPadding + 2 * sizeof(uint64_t);
```

Comment:
We have already added the kPadding in PaddedAllocationSize, it's unnecessary to add it again.